packages/live-status-gateway/src/collections/adLibActions.ts

```diff
@@ -34,15 +35,17 @@ export class AdLibActionsHandler
 	async update(source: string, data: Map<PartInstanceName, DBPartInstance | undefined> | undefined): Promise<void> {
 		this._logger.info(`${this._name} received partInstances update from ${source}`)
 		const prevRundownId = this._curRundownId
-		this._curRundownId = data ? unprotectString(data.get(PartInstanceName.current)?.rundownId) : undefined
+		const prevCurPartInstance = this._curPartInstance
+		this._curPartInstance = data ? data.get(PartInstanceName.current) ?? data.get(PartInstanceName.next) : undefined
+		this._curRundownId = this._curPartInstance ? unprotectString(this._curPartInstance.rundownId) : undefined
 
 		await new Promise(process.nextTick.bind(this))
 		if (!this._collection) return
 		if (!this._publication) return
-		if (prevRundownId !== this._curRundownId) {
+		if (!(prevRundownId === this._curRundownId && prevCurPartInstance === this._curPartInstance)) {
```

Comment:
```suggestion
		if (prevRundownId !== this._curRundownId || prevCurPartInstance !== this._curPartInstance)) {
```
I think this is easier to quickly interpret

Reply:
Should this be comparing object references of `_curPartInstance`, or should it do a deeper equality check?
It looks like it only really needs the `PartId` of the PartInstance. Should it store and compare just that value instead to reduce the amount of invalidations?